### ISSUE TITLE
Provide a way to customize gRPC content logging

### ISSUE THREAD

**Issue by 0x1306e6d** [line/armeria#5777]
Hello,

When using `LoggingService` with a gRPC service, an escaped content is logged when non-ASCII string is included in its message:

```
content=CompletableRpcResponse{foo {
  id: "J9CfU0LU"
  name: "\354\225\210\353\205\225\355\225\230\354\204\270\354\232\224"
}
}
```

I guess that this is the default behavior of protobuf-java's `toString()`. There's an [option](https://github.com/protocolbuffers/protobuf/blob/8ed10a9a464c4fdb9d77fbbd082763d7c29c26a2/java/core/src/main/java/com/google/protobuf/TextFormat.java#L163) not to escape, so I could address by adding `contentSanitizer` to use the custom `TextFormat` only when the content type is protobuf `Message`:

```java
LogFormatter.builderForText()
            .contentSanitizer(
                    (requestContext, o) -> {
                        if (o instanceof CompletableRpcResponse res) {
                            final Object result = res.getNow(null);
                            if (result instanceof Message message) {
                                return TextFormat.printer()
                                                 .escapingNonAscii(false)
                                                 .printToString(message);
                            }
                        }
                        return o.toString();
                    })
            .build();
```

For gRPC (Thrift also I guess), `Logging{Client|Service}` simply calls `toString()` of a `CompletableRpcResponse` and there's a weak support for handling its content for logging. It'd be useful if we provide a way to handle the content.

**Comment by trustin** [comment]
Would it be a good idea for us to disable the escape option by default? Is there any risk of doing so? I think it's perhaps a good idea to escape the control characters though.

**Comment by 0x1306e6d** [comment]
I agree that disabling by default has a risk. I think that keeping the default behavior and providing an option to control would be better so that users can take the risk.

**Comment by trustin** [comment]
I'm just trying to assess the risk. What risk would it have?
